src/custom/state/orders/mocks.ts

```diff
@@ -0,0 +1,180 @@
+
+  const orderType = orderN % 2 ? OrderKind.BUY : OrderKind.SELL
+
+  const summary = `Order ${orderType.toUpperCase()} ${+sellAmount / 10 ** sellToken.decimals} ${
+    sellToken.symbol
+  } for ${+buyAmount / 10 ** buyToken.decimals} ${buyToken.symbol} fulfilled`
+
+  return {
+    id: generateOrderId(orderN), // Unique identifier for the order: 56 bytes encoded as hex without 0x
+    owner: owner.replace('0x', ''),
+    status: OrderStatus.PENDING,
+    creationTime: new Date().toISOString(),
+    summary, // for dapp use only, readable by user
+    sellToken: sellToken.address.replace('0x', ''), // address, without '0x' prefix
+    buyToken: buyToken.address.replace('0x', ''), // address, without '0x' prefix
+    sellAmount, // in atoms
+    buyAmount, // in atoms
+    // 4 - 6min
+    validTo: Date.now() / 1000 + randomIntInRangeExcept(240, 360), // uint32. unix timestamp, seconds, use new Date(validTo * 1000)
+    appData: 1, // arbitrary identifier sent along with the order
+    feeAmount: (1e18).toString(), // in atoms
+    orderType,
+    partiallyFillable: false,
+    signature: (orderN++).toString().repeat(65 * 2) // 65 bytes encoded as hex without `0x` prefix. v + r + s from the spec
+  }
+}
+
+// add more orders if less than minPendingOrders currently
+const useAddOrdersOnMount = (minPendingOrders = 5) => {
+  const { account, chainId, library } = useActiveWeb3React()
+
+  const pendingOrders = usePendingOrders({ chainId })
+
+  // ref, so we don't rerun useEffect
+  const pendingOrdersRef = useRef(pendingOrders)
+  pendingOrdersRef.current = pendingOrders
+
+  const addOrder = useAddPendingOrder()
+
+  const lists = useSelectedTokenList()
+
+  useEffect(() => {
+    const addNOrders = (ordersNum: number) => {
+      if (!account || !chainId || !library) return
+      const tokenMap = lists[chainId]
+
+      const tokenList = Object.values(tokenMap)
+      if (tokenList.length === 0) return
+
+      const newTempOrders = Array.from({ length: ordersNum }, () => {
+        const [sellToken, buyToken] = getTwoRandomElementsFromArray(tokenList)
+
+        return generateOrder({
+          owner: account,
+          sellToken: sellToken.tokenInfo,
+          buyToken: buyToken.tokenInfo
+        })
+      })
+
+      batch(() => {
+        newTempOrders.forEach(order => {
+          addOrder({ order, id: order.id, chainId })
+        })
+      })
+    }
+    ;(window as any).addNOrders = addNOrders
+
+    // don't just keep adding orders when there's already enough pending
+    if (pendingOrdersRef.current.length >= minPendingOrders) return
+
+    addNOrders(10)
+  }, [account, addOrder, chainId, library, lists, minPendingOrders])
+}
+
+const useFulfillOrdersRandomly = (interval = 20000 /* ms */) => {
+  const { chainId } = useActiveWeb3React()
+  const pendingOrders = usePendingOrders({ chainId })
+
+  // ref, so we don't rerun useEffect
+  const pendingOrdersRef = useRef(pendingOrders)
```

Comment:
I don't fully understand

Reply:
`pendingOrders` is a dependency is `useEffect`